File: src/com/castlefrog/agl/TurnType.java
package com.castlefrog.agl;

/**
 * Indicates the type of turn
 * mechanic the game uses.
 */
public enum TurnType {
    /** next player to move decided at random */
    RANDOM,
    /** players move in order */
    SEQUENTIAL,
    /** player move order random */
    RANDOM_ORDER,
    /** player move order decided by bidding */
    BIDDING,
    /** players move at same time */
    SIMULTANEOUS
    //SIMULTANEOUS_PRIORITY,
}

File: api/src/main/java/com/castlefrog/agl/TurnType.kt
package com.castlefrog.agl

/**
 * Indicates how agent turns are decided.
 */
enum class TurnType {
    /** next player to move decided at random  */
    RANDOM,
    /** players move in order  */
    SEQUENTIAL,
    /** player move order randomized  */
    RANDOM_ORDER,
    /** next player to move decided by bidding  */
    BIDDING,
    /** players move at same time  */
    SIMULTANEOUS,
    /** players move at same time, conflicts resolved with priority  */
    SIMULTANEOUS_PRIORITY
}
